pkg/versiongetter/general.go

```diff
@@ -65,3 +67,19 @@ func itemNumPerPage(limit, filterNum int) int {
 	}
 	return ghMaxPerPage
 }
+
+// log the GitHub API rate limit info
+func logGHRateLimit(logE *logrus.Entry, resp *github.Response) {
+	logE.WithFields(logrus.Fields{
+		"gh_rate_limit":     resp.Rate.Limit,
+		"gh_rate_remaining": resp.Rate.Remaining,
+	}).Info("GitHub API rate limit info")
+}
+
+// fuzzy-finder's output will overwrite the log, add fields to original logE
+func addRteLimitInfo(logE *logrus.Entry, resp *github.Response) {
+	*logE = *logE.WithFields(logrus.Fields{
+		"gh_rate_limit":     resp.Rate.Limit,
+		"gh_rate_remaining": resp.Rate.Remaining,
+	})
+}
```

Comment:
```suggestion
func addRateLimitInfo(logE *logrus.Entry, resp *github.Response) *logrus.Entry {
	return logE.WithFields(logrus.Fields{
		"gh_rate_limit":     resp.Rate.Limit,
		"gh_rate_remaining": resp.Rate.Remaining,
	})
}
```

I think this is more natural.

```go
logE = addRateLimitInfo(logE, resp)
```

Reply:
```go
logE = addRateLimitInfo(logE, resp)
```
This statement generate a derived logE and assign its pointer to a local variable.

This will not change its orginal value.

see this example in [playground](https://go.dev/play/p/s5aRHw0q-Wk)

```go
*logE = *addRateLimitInfo(logE, resp)
```
What do you think of editing it like this?

Reply:
You don't need to change the original value.
You just create a new *logrus.Entry and use it.
I think this is the design of logrus.

Reply:
The reason why I want to change its original value is the fuzzy-finder's output will overwirte the log output.

See this [code](https://github.com/aquaproj/aqua/blob/main/pkg/versiongetter/fuzzy_getter.go#L35C2-L53) 

`List` is invoked before openning fuzzy-finder, if log something in `List` , it will be overwritten by fuzzy-finder.

```
List -> print log  -> open fuzzy-finder -> print content (overwrite the log)
```

Reply:
I record this to help you to understand what I'm describing.

![](https://raw.githubusercontent.com/dreamjz/pics/main/pics/2023/202311142333264.gif)

Reply:
Thank you for your explanation. I understood the issue, but I don't want to change `*logrus.Entry`.
This is tricky.
I don't think the information about API rate limit is so important that we need to output it with the tricky way unless API rate limit error occurs.